pkgs/development/python-modules/scalene/default.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, setuptools-scm
+, cloudpickle
+, cython
+, jinja2
+, psutil
+, pynvml
+, pythonOlder
+, rich
+}:
+
+buildPythonPackage rec {
+  pname = "scalene";
+  version = "1.5.35";
+  pyproject = true;
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-3s0KN06RivCN4G5/LZf+r7GtAlKwqRuZwUtrY2T2Bew=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-warn "pynvml>=11.0.0,<11.5" "pynvml~=11.5.0"
+    substituteInPlace requirements.txt \
+      --replace-warn "pynvml~=11.0.0" "pynvml~=11.5.0"
+  '';
+
+  nativeBuildInputs = [
+    cython
+    setuptools
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    cloudpickle
+    jinja2
+    psutil
+    pynvml
+    rich
+  ];
+
+  pythonImportsCheck = [ "scalene" ];
+
+  meta = with lib; {
+    description = "Scalene: A high-resolution, low-overhead CPU, GPU, and memory profiler for Python with AI-powered optimization suggestions";
```

```suggestion
    description = "High-resolution, low-overhead CPU, GPU, and memory profiler for Python with AI-powered optimization suggestions";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

---

Nice. Thank you.